src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs

```diff
@@ -339,6 +339,8 @@ public void SetData(string format, Object data, bool autoConvert)
         {
             ArgumentNullException.ThrowIfNull(format);
 
+            ArgumentNullException.ThrowIfNull(data);
```

Comment:
Digression: should we remove the comment from the remarks:

https://github.com/dotnet/wpf/blob/821aa2440005f3b8911d6beb19d798155c3bd931/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs#L334-L336

Recent versions of .NET do not honor Code Access Security (CAS).

Reply:
Also, I think we need to maintain parity between the implementations of the `SetData` functions. 
The check for `null` can be done after the `format` empty check to align with the same behavior.